### PR TITLE
Edituser

### PR DESCRIPTION
**Pull request by sohilpandya**
Hi @minaorangina,

I've added the functionality for lecturer to be able to view all the users that are part of a given module and also remove users from the given module. 

To view this page, lecturer has to the following label in module details.

screen shot 2016-08-15 at 19 57 21

### CODE REVIEW

**Comment by sohilpandya** [comment]
@minaorangina I was not sure about that either. current function deletes the user from the module.

I thought about this, and I feel that it is more helpful to have this over the complete deletion of a user, as a complete deletion of a user should happen under a super admin functionality. (potentially in the future)

should we ask Stian and see what he would prefer?

File: test/unit/actions/module.test.js [hunk 1]
```diff
@@ -77,3 +142,59 @@ test('closeQuiz creates the correct action', (t) => {
     const actual = deepFreeze(actions.closeQuiz());
     t.deepEqual(actual, expected);
 });
+
+
+//
+// GET MODULE  ACTIONS
```
Comment (minaorangina) [inline test/unit/actions/module.test.js]: This comment isn't right - should be `REMOVE MODULE MEMBER ACTIONS`

File: src/js/actions/module.js [hunk 2]
```diff
@@ -55,3 +63,76 @@ export const getModuleFailure = (error) => ({
     type: GET_MODULE_FAILURE,
     error
 });
+
+//
+// GET MODULE USERS actions
```
Comment (minaorangina) [inline src/js/actions/module.js]: --> `GET MODULE MEMBERS`

File: src/js/components/module/details.js [hunk 3]
```diff
@@ -10,7 +11,9 @@ const Details = ({ name, module_id, num_enrolled, trophies, medals }) => {
 
                 <h2>{ name }</h2>
                 <h5>{ module_id }</h5>
+                <Link to={ `${module_id}/users` }>
```
Comment (minaorangina) [inline src/js/components/module/details.js]: I guess this endpoint should be `members` as well

File: src/js/components/module/members.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import React, { PropTypes } from 'react'; //eslint-disable-line no-unused-vars
+import Spinner from '../general/spinner';
+import Nav from '../general/nav';
+
+const Users = ({ users, isFetchingModuleMembers, isRemovingMember, username, handleRemovingUser, location }) => {
```
Comment (minaorangina) [inline src/js/components/module/members.js]: component name should be `Members`, and the handler should be `handleRemovingMember`

File: src/js/containers/module-members.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+const mapStateToProps = (state) => ({
+    users: state.module.users,
+    isFetchingModuleMembers: state.module.isFetchingModuleMembers,
+    isRemovingMember: state.module.isRemovingMember,
+    username: state.user.username
+});
+
+const mapDispatchToProps = (dispatch) => ({
+
+    handleRemovingUser: (module_id, user_id) => {
```
Comment (minaorangina) [inline src/js/containers/module-members.js]: also `handleRemovingMember`

File: src/js/containers/module-members.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { connect } from 'react-redux';
+import Members from '../components/module/members';
+import { removeModuleMember } from '../actions/module';
+
+const mapStateToProps = (state) => ({
+    users: state.module.users,
+    isFetchingModuleMembers: state.module.isFetchingModuleMembers,
```
Comment (minaorangina) [inline src/js/containers/module-members.js]: For the sake of brevity, if we're saying `isRemovingMember`, may as well be consistent and say `isFetchingMembers`

File: src/js/reducers/module.js [hunk 7]
```diff
@@ -5,8 +5,11 @@ const initialState = {
     module: undefined,
     quizzes: undefined,
     isFetchingModule: false,
+    isFetchingModuleMembers: false,
     error: undefined,
-    isQuizOpen: false
+    isQuizOpen: false,
+    users: undefined,
```
Comment (minaorangina) [inline src/js/reducers/module.js]: this should now be `members` as well

File: test/utils/reducer-fixtures.js [hunk 8]
```diff
@@ -56,7 +56,9 @@ export const module = {
     quizzes: undefined,
     isFetchingModule: false,
     error: undefined,
-    isQuizOpen: false
+    isQuizOpen: false,
+    users: undefined,
```
Comment (minaorangina) [inline test/utils/reducer-fixtures.js]: Same goes for the fixtures file

**Pull request merged**